### ISSUE TITLE
Detaching a Script from a Node Produces Neither Warning nor Confirmation of the Action

### ISSUE THREAD

**Issue by jtwine** [godotengine/godot#80308]
### Godot version

v4.1.1.stable.official [bd6af8e0e]

### System information

Godot v4.1.1.stable - Windows 10.0.22621 - Vulkan (Compatibility) - NVIDIA GeForce GTX 1060 (NVIDIA; 30.0.14.7311) - Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz (12 Threads)

### Issue description

Many destructive actions provide you with a confirmation dialog.  However, the "detach script" function does not seem to be one of them.  The script will get _silently_ detached from the selected node _without any warning or confirmation_.

Suggest that either a warning or a confirmation of the action be provided.  This is because an errant click can cause detachment and you might not notice it quickly enough to easily perform an Undo to resolve the issue.

### Steps to reproduce

Simply press **[Detach the script from the selected node.]** on a node with a script attached.

### Minimal reproduction project

N/A

**Comment by Ymanawat** [comment]
I don't think it's necessary, as there is no loss from this action. We can simply undo the action or reattach the script since it hasn't been deleted. @Calinou could you please confirm if this is required?

**Comment by jtwine** [comment]
Just remember that there are other actions in Godot that result in confirmation dialogs (e.g. deletion of a Node, deletion of an Animation) and they too are reversible via **Undo** actions -- but they still have confirmation dialogs.  Simply being able to Undo an action should not by itself be a good enough reason, IMHO.  Otherwise it would be a _good enough_ reason to go and remove the existing confirmation dialogs for those other actions, simply because they are **Undo**able, right?

Remember that consistent behavior _goes a long way_ when dealing with UI/UX issues.  

The other thing to remember is that this particular action is a "silent" one.  If the user accidently disconnects a script, _they may have no idea that they did it_, and may be confused as to why something suddenly stopped working correctly in their game later on.  (Although that might just warrant a confirmation dialog.) 

-=- James.

**Comment by Ymanawat** [comment]
I agree. I thought that there would be no data loss by this action. But, when we export some variables or an array of variables and then disconnect the script, all the data gets lost. Even after undoing, we get the default value we set in the script. So, I think it's a good reason to do it.

**Comment by jtwine** [comment]
...if only there was a way to reopen the issue... :)

**Comment by Calinou** [comment]
> I agree. I thought that there would be no data loss by this action. But, when we export some variables or an array of variables and then disconnect the script, all the data gets lost. Even after undoing, we get the default value we set in the script. So, I think it's a good reason to do it.

In this case, it makes sense to display a confirmation dialog if the script has exported properties with non-default values – or better, attempt to restore the custom property values.

**Comment by andrewdirisio** [comment]
Chiming in here having just gone through this myself. I too believe a confirmation is needed when detaching a script. A toast style notification would be an upgrade, but still easy to overlook when looking across 2 or more monitors. I would hope for something more concrete since these actions are often unintended, but can easily snowball.

**The previous point made about this being a _silent action_ is a very important one, and something that I don't believe is fixed even if the exported properties are restored, as going back to that state also undoes any other actions made since then. It may not even be on the list of history states depending on when you notice, or even more, you save and close the project.** 

In my case, I unknowingly detached the script on my camera in the scene I was working on. It had an exported array of variables with all my different camera shake types, each with their own list of settings dialed in. The scene I was working on was my main level, so my history panel was quickly filled with all different parts and pieces being added in and all their moves. It wasn't until I went to test the level did I see an error and realize what happened. Now I'm stuck between undoing the level I just built or setting my camera up from scratch.

It was a very disheartening lesson to learn. One I will do my best to take precaution against going forward, however I think the need for this to be an option is very real, and could serve to cast a wider net of protection than hindrance to the user experience.
